generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management-update.component.ts.ejs

```diff
@@ -344,18 +357,16 @@ _%>
     }
 
     <%_ if (selectableEntitiesType) { _%>
-    trackById(index: number, item: <%= selectableEntitiesType %>): any {
-        return item.id;
+    trackById(index: number, item: <%= selectableEntitiesType %>): <%= selectableEntitiesIdType %> {
+        return item.id!;
     }
     <%_ } _%>
 
     <%_ if (selectableManyToManyEntitiesType) { _%>
     getSelected(selectedVals: <%= selectableManyToManyEntitiesType %>[], option: <%= selectableManyToManyEntitiesType %>): <%= selectableManyToManyEntitiesType %> {
-        if (selectedVals) {
```

Comment:
some problem with this change
See https://github.com/jhipster/generator-jhipster/issues/12928